src/pastelid/ed.h

```diff
@@ -415,4 +415,24 @@ namespace ed_crypto {
     using key_dh448 = key<EVP_PKEY_X448>;
     using key_dsa25519 = key<EVP_PKEY_ED25519>;
     using key_hd25519 = key<EVP_PKEY_X25519>;
+
+    static bool isValidPassphrase(std::string pastelid, SecureString strKeyPass)
+    {
+        try {
+            //Get pastelkeyfile
+            fs::path pathPastelKeys(GetArg("-pastelkeysdir", "pastelkeys"));
+            pathPastelKeys = GetDataDir() / pathPastelKeys;
+
+            if (!fs::exists(pathPastelKeys)){
+                return false;
+            }
+
+            fs::path pathPastelKeyFile = pathPastelKeys / pastelid;
```

Comment:
check if pathPastelKeyFile exists?
then you can remove above checking of "pathPastelKeys" exists